libraries/common/list.effekt

```diff
@@ -601,6 +601,40 @@ def zip[A, B](left: List[A], right: List[B]): List[(A, B)] = {
   go(Nil(), left, right)
 }
 
+/// Produce a list of These from two lists.
+/// The length is the maximum of lenghts of the two lists.
+///
+/// Examples:
+/// ```
+/// > zipLongest([1, 2, 3], [100, 200, 300])
+/// [These(1, 100), These(2, 200), These(3, 300)]
+///
+/// > zip([1, 2, 3], Nil[Int]())
+/// [This(1), This(2), This(3)]
+///
+/// > zip(Nil[Int](), [1, 2, 3])
+/// [That(1), That(2), That(3)]
+///
+/// > zip([1, 2, 3], [42])
+/// [These(1, 42), This(2), This(3)]
+/// ```
+///
+/// O(N)
+def zipLongest[A, B](left: List[A], right: List[B]): List[These[A, B]] = {
```

Comment:
Bikeshed: [Unison](https://share.unison-lang.org/@unison/base/code/releases/4.0.0/latest/terms/data/List/align) calls this `align` which is a bit nicer, perhaps?
(ditto for streams)

Reply:
We could also "just" call it `zip` and rely on the overload resolution ;)